app/controllers/application_controller.rb

```diff
@@ -1,9 +1,16 @@
 class ApplicationController < ActionController::Base
   include Pundit::Authorization
-  before_action :authenticate_user!, :set_users
+  before_action :authenticate_user!, :set_users, :set_notifications
   protect_from_forgery
 
   def set_users
     @users = User.all
   end
+
+  def set_notifications
+    new_announcement_date_time = Rails.cache.read('new_announcement')
+    last_viewed_announcements_date_time = cookies[:announcements_viewed]
+    new_announcement_notification = new_announcement_date_time && (!last_viewed_announcements_date_time || new_announcement_date_time > last_viewed_announcements_date_time)
```

Comment:
Here's where the magic now happens. If there has been an announcement within the past 10 days _and_ the user hasn't viewed the announcements page since that new announcement the notification will appear. It will disappear after the user views the announcements page and updates their "announcements_viewed" cookie time.